pkg/virt-controller/watch/clone/vm-target.go

```diff
@@ -57,18 +62,27 @@ func generatePatches(source *k6tv1.VirtualMachine, cloneSpec *clonev1alpha1.Virt
 	return patches
 }
 
-func generateMacAddressPatches(interfaces []k6tv1.Interface, newMacAddresses map[string]string) (patches []string) {
-	const macAddressPatchPattern = `{"op": "replace", "path": "/spec/template/spec/domain/devices/interfaces/%d/macAddress", "value": "%s"}`
-
+func generateMacAddressPatches(interfaces []k6tv1.Interface, newMacAddresses map[string]string) (patches []string, err error) {
+	// If the newMacAddresses map is nil, no patches would be generated.
+	if newMacAddresses == nil {
+		return
+	}
+	patchSet := patch.New()
```

Comment:
I would define this outside this function and only add here the patches for the mac address